libnavigation-ui/src/main/java/com/mapbox/navigation/ui/route/MapRouteLine.kt

```diff
@@ -1339,62 +1458,46 @@ internal class MapRouteLine(
             }
         }
 
-        val getDistanceBetweenPoints: (firstPoint: Point, secondPoint: Point) -> Double =
-            { firstPoint: Point, secondPoint: Point ->
-                TurfMeasurement.distance(firstPoint, secondPoint, TurfConstants.UNIT_METERS)
-            }.memoize(POINT_DISTANCE_CALCULATION_FUN_CACHE_SIZE)
-
-        val getReversedCoordinates: (line: LineString) -> List<Point> = { line: LineString ->
-            line.coordinates().reversed()
-        }.memoize(1)
-
-        fun calculatePreciseDistanceTraveledAlongLine(
-            line: LineString,
-            targetDistance: Float,
-            targetPoint: Point
-        ): Double {
-            val linePoints = getReversedCoordinates(line)
-            var runningDistance = 0.0
-            var lastPointIndex = 0
-
-            for (currentIndex in linePoints.indices) {
-                if (currentIndex + 1 < linePoints.size) {
-                    val nextSectionDistance = getDistanceBetweenPoints(
-                        linePoints[currentIndex],
-                        linePoints[currentIndex + 1]
-                    )
-                    if (runningDistance + nextSectionDistance > targetDistance) {
-                        val currentPointToTargetPointDist =
-                            TurfMeasurement.distance(
-                                linePoints[currentIndex],
-                                targetPoint,
-                                TurfConstants.UNIT_METERS
-                            )
-
-                        if (currentPointToTargetPointDist > nextSectionDistance) {
-                            runningDistance += nextSectionDistance
-                            lastPointIndex = currentIndex + 1
-                        } else {
-                            lastPointIndex = currentIndex
-                        }
-                        break
-                    } else {
-                        runningDistance += nextSectionDistance
-                    }
-                } else {
-                    lastPointIndex = currentIndex
-                }
+        fun calculateGranularDistances(points: List<Point>): RouteLineGranularDistances {
+            var distance = 0.0
+            val indexArray = SparseArray<RouteLineDistancesIndex>(points.size)
+            for (i in (points.size - 1) downTo 1) {
+                val curr = points[i]
+                val prev = points[i - 1]
+                distance += calculateDistance(curr, prev)
+                indexArray.append(i - 1, RouteLineDistancesIndex(prev, distance))
             }
+            indexArray.append(
+                points.size - 1,
+                RouteLineDistancesIndex(points[points.size - 1], 0.0)
+            )
```

Comment:
Same as above re: extracting to `private` `fun`ctions so it's easier to read and understand